pkg/cmd/roachtest/tests/follower_reads.go

```diff
@@ -884,25 +899,57 @@ func parsePrometheusMetric(s string) (*prometheusMetric, bool) {
 	}, true
 }
 
-// runFollowerReadsMixedVersionSingleRegionTest runs a follower-reads test while
-// performing a cluster upgrade. The point is to exercise the closed-timestamp
-// mechanism in a mixed-version cluster. Running in a single region is
-// sufficient for this purpose; we're not testing non-voting replicas here
-// (which are used in multi-region tests).
+// runFollowerReadsMixedVersionSingleRegionTest runs a follower-reads test in a
+// single region while performing a cluster upgrade. The point is to exercise
+// the closed-timestamp mechanism in a mixed-version cluster. Running in a
+// single region is sufficient for this purpose; we're not testing non-voting
+// replicas here (which are used in multi-region tests).
 func runFollowerReadsMixedVersionSingleRegionTest(
 	ctx context.Context, t test.Test, c cluster.Cluster,
 ) {
-	mvt := mixedversion.NewTest(ctx, t, t.L(), c, c.All(),
-		// The http requests to the admin UI performed by the test don't play
-		// well with secure clusters. As of the time of writing, they return
-		// either of the following errors:
-		//  tls: failed to verify certificate: x509: “node” certificate is not standards compliant
-		//  tls: failed to verify certificate: x509: certificate signed by unknown authority
-		//
-		// Disable secure mode for simplicity.
-		mixedversion.ClusterSettingOption(install.SecureOption(false)),
-	)
 	topology := topologySpec{multiRegion: false}
+	runFollowerReadsMixedVersionTest(ctx, t, c, topology, exactStaleness)
+}
+
+// runFollowerReadsMixedVersionGlobalTableTest runs a multi-region follower-read
+// test with a region-survivable global table while performing a cluster upgrade.
+// The point is to exercise global tables in a mixed-version cluster.
+func runFollowerReadsMixedVersionGlobalTableTest(
+	ctx context.Context, t test.Test, c cluster.Cluster,
+) {
+	topology := topologySpec{
+		multiRegion: true,
+		locality:    global,
+		survival:    region,
+	}
+	runFollowerReadsMixedVersionTest(ctx, t, c, topology, strong,
+		// Disable fixtures because we're using a 6-node, multi-region cluster.
+		mixedversion.NeverUseFixtures,
+		// Use a longer upgrade timeout to give the migrations enough time to finish
+		// considering the cross-region latency.
+		mixedversion.UpgradeTimeout(60*time.Minute),
```

Comment:
Curious how long it takes today?

Reply:
The entire test takes about 80 minutes. I did see one instance where an upgrade migration timed out after 10m, so I'm conservatively bumping this up to give migrations more time.